### DISCUSSION TITLE
Custom sounds

### DISCUSSION THREAD

**Discussion by zumzum360** [FrostCo/AdvancedProfanityFilter#528] [Ideas]
can you have a feature that you can put custom filter sound effects?

**Comment by richardfrost** [comment]
Interesting idea @zumzum360, I'll add it to the list and look into it when I get some more time. Thanks!

**Comment by zumzum360** [comment]
...any updates now?

**Reply by richardfrost** [reply]
I think it should be possible to add a custom sound, with some limitations, and I'll let you know when I have something ready to test. Thanks for the reminder on this.

**Comment by zumzum360** [comment]
you're welcome

**Comment by zumzum360** [comment]
any progress now?

**Reply by richardfrost** [reply]
@zumzum360 Unfortunately I've been completely swamped with other things, and I haven't had a chance to start on it. I am hoping things calm down by next week and then I'll see if I can get it added sometime next week.

**Reply by zumzum360** [reply]
okay

**Comment by zumzum360** [comment]
okay

**Comment by zumzum360** [comment]
any updates now?

**Reply by richardfrost** [reply]
I'm working through it. Made it a bit over half way through, so it should be included in the next version, hopefully sometime in the coming week.

**Reply by zumzum360** [reply]
okay

**Comment by zumzum360** [comment]
any updates now?

**Reply by richardfrost** [reply]
Not yet. I haven't had a chance to finish testing, but it should be pretty close once I get some more time to complete testing. I am pretty busy this weekend, but I am hoping to find some time to work on it. Unfortunately I had to make some other changes to the app to support the custom sounds, so I have to do a bit more testing to feel confident nothing else broke, but the custom audio does indeed work, so it shouldn't be too long now.

**Comment by zumzum360** [comment]
okay

**Comment by richardfrost** [comment]
@zumzum360 I still have a few more things I need to test outside of the custom sound, but if you want to take this build for a test feel free to use [these instructions](https://github.com/FrostCo/AdvancedProfanityFilter/wiki/Load-Unpacked-Extension) with this build: [chrome-mv3-v3.5.1.1.zip](https://github.com/user-attachments/files/16420338/chrome-mv3-v3.5.1.1.zip).

As soon as I'm done with the rest of the testing I'll get a new version that includes this, so it should be done pretty soon. With this unpacked extension, you'll need to import your settings (or just start from scratch). To add a custom filler sound, you can select the "Custom" option from the filler audio dropdown, and then click the "Upload" button that appears. Only MP3 files are supported, and I would strongly recommend using a small file (try for < `500kb`). Due to how storage works with an extension, I'm limiting it to a single custom sound, so if you decide you want a different one you'll need to repeat the "Upload" process.

**Reply by richardfrost** [reply]
@zumzum360 This feature has been added to [version 3.6.0](https://github.com/FrostCo/AdvancedProfanityFilter/releases/tag/3.6.0). Feel free to share any feedback here.

**Comment by zumzum360** [comment]
thank you very much

**Reply by richardfrost** [reply]
You're welcome! Sorry again that it took so long.
